cypress/integration/diagrams.spec.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+describe('renders diagram', () => {
+  beforeEach(() => {
+    cy.visitTestEditor()
+  })
+
+  it('markmap', () => {
```

Comment:
"Renders diagram markmap" is not a valid sentence. The describe+it calls should build a valid sentence like "Short code for pdf renders a plain link". That's why it's called "it" e.g. "It renders a plain link".
What about
"Diagram embeddings" + "for markdown renders a svg".

Reply:
Well, "Short code for pdfs" isn't a sentence as well, just to name one example.
A complete sentence seems a bit verbose. What about the following schema?

Diagram embedding renders correctly: <markmap|vega-lite|...>

Reply:
Well that's true. Okay, how about this: Describe blocks don't have to be part of the senteces, but should specify the parent describe block. But I insist, that the it-calls build correct senteces. It is a convention in testing.

Reply:
It's important to have a good structure.
And a complete sentence might read good in single occasions and when you read it alone standing. But if you have dozens of log lines - like "Component xabc renders as expected", "Component ydfghj renders as expected" - it's much harder to get an overview of the actual component tested, especially as similar words are not aligned beneath each other.

I therefore stay at the position that test labels like "renders as expected: xabc", "renders as expected: ydfghj" fulfill the same purpose while being easier to read.

Reply:
I'm seeing this the opposite way. For me a senteces if much better readable. Especially in the output.

Sry but then I can't approve this, if we don't agree on a syntax, because then some tests are names like you want and some like I want and maybe @DerMolly also has another opinon.

Reply:
okay what about
describe: `rendering diagram`
	it: `markmap works`

Reply:
@mrdrogdrog I don't understand why suddenly now you want to enforce your way. If we had agreed to always use such a structure or if every other test follows the same schema, it would be a different thing. But: From 131 cypress tests being in main branch less than the half - more exact 60 - are written as a sentence.

Moreover, I'll show you why I don't agree on sentences. Look at the image below. On a quick glimpse which of the tests could have failed you just see "shows correct line number" and have to read the beginning of your sentence above that.

![image](https://user-images.githubusercontent.com/52606896/108052415-65a3ef80-704c-11eb-868a-a766b3d8989e.png)

Reply:
Well. It's a pity that only jest seems to follow the convention.

We never agreed on a syntax, therefore I started to use the conventional syntax. This is why I started "now". 
Anyway, we should agree on something.

Imho the test case name should include the expectation not the object that is tested. That's what a describe block is for.

Reply:
like `"Diagram embeddings" + "renders markmap"`

Reply:
> like `"Diagram embeddings" + "renders markmap"`

Did that.